app/bundles/LeadBundle/Tests/EventListener/WebhookSubscriberTest.php

```diff
@@ -186,4 +192,38 @@ public function testOnCompanySaveAndDelete()
         $dispatcher->dispatch($event, LeadEvents::COMPANY_POST_SAVE);
         $dispatcher->dispatch($event, LeadEvents::COMPANY_POST_DELETE);
     }
+
+    public function testOnSegmentChangeWithArrayContact(): void
+    {
+        $changeEvent = $this->createMock(ListChangeEvent::class);
+
+        $contact = ['id' => 1];
+
+        $changeEvent->method('getLeads')->willReturn([$contact]);
+        $changeEvent->method('getLead')->willReturn(null);
+        $changeEvent->method('wasAdded')->willReturn(true);
+
+        $leadModel    = $this->createMock(LeadModel::class);
+        $webhookModel = $this->createMock(WebhookModel::class);
+
+        $leadModel->expects($this->once())
+            ->method('getEntity')
+            ->with($this->equalTo($contact['id']))
+            ->willReturn($contact);
```

Comment:
This should return the entity, right? Not the array. I guess it doesn't matter for the test but it could be confusing for whoever will be checking how it works.